src/IIIFingest/asset.py

```diff
@@ -116,13 +137,93 @@ def from_file(cls, filepath, **kwargs):
             metadata=metadata,
         )
 
+    @classmethod
+    def from_fileobj(cls, fileobj: BinaryIO, **kwargs) -> Asset:
+        """
+        Constructs an Asset from a file-like object.
+
+        Args:
+            fileobj:
+                A file-like object from which the asset will be generated.
+            **kwargs:
+                Several specific kwargs can be passed to the function, but if
+                they are omitted they will be inferred from the file. These
+                optional arguments are specified below.
+            asset_id:
+                A unique identifier for the asset. If none is specified, this
+                attribute of the Asset object output will be None.
+            width:
+                Width of the image in pixels. Ignored if height is not present.
+            height:
+                Height of the image in pixels. Ignored if width is not present.
+            format:
+                Mime type of the image. If not specified, this property can be
+                inferred from the content_type attribute on a Django
+                UploadedFile object, or directly from the file.
+            extension:
+                File extension for the image. If not specified, it is inferred
+                from the mime type.
+            label:
+                An optional label for the image. If none is specified, then the
+                asset_id will be used as a label.
+            metadata:
+                Metadata dictionary to be assigned to the asset.
+
+        Returns:
+            A newly constructed asset object.
+        """
+        asset_id = kwargs.get("asset_id")
+
+        if kwargs.get("width") and kwargs.get("height"):
+            width = kwargs["width"]
+            height = kwargs["height"]
+        else:
+            width, height = get_image_size(fileobj)
+
+        if kwargs.get("format"):
+            format = kwargs.get("format")
+        elif hasattr(fileobj, 'content_type'):
+            # Django UploadedFile objects have a content type attribute
+            # that can be used here
+            format = fileobj.content_type
+        else:
+            # Get the mime type using libmagic, ensuring that the file pointer
+            # is at the top of the file.
+            validator = magic.Magic(mime=True, uncompress=True)
+            fileobj.seek(0)
+            format = validator.from_buffer(fileobj.read(1024))
```

Comment:
This is just a general question for my own edification: do we know if `1024` is the right number to use here or if it's documented anywhere? 

Libmagic seems to be rather esoteric and unclear on this point. The https://github.com/ahupp/python-magic readme, recommends `2048` bytes, but that seems to be for the most general case where it could be any file format. We know it's going to be some kind of image format, so  I'm going to guess 1024 bytes will be sufficient and that your tests confirmed that to be the case.

Reply:
I used 1024 since that's what an example I was looking at used, so 2048 as recommended seems reasonable. I think the most important thing is to limit it in some way, and I'm happy to err on the side of reading more, just in case.

Reply:
Yeah I came across an example somewhere that used 1024, and then saw the recommendation in the readme. I guess it wouldn't hurt to err on the side of reading a few more bytes.